app/models/concerns/status/fetch_replies_concern.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Status::FetchRepliesConcern
+  extend ActiveSupport::Concern
+
+  # enable/disable fetching all replies
+  FETCH_REPLIES_ENABLED = ENV.key?('FETCH_REPLIES_ENABLED') ? ENV['FETCH_REPLIES_ENABLED'] == 'true' : true
+
+  # debounce fetching all replies to minimize DoS
+  FETCH_REPLIES_COOLDOWN_MINUTES = (ENV['FETCH_REPLIES_COOLDOWN_MINUTES'] || 15).to_i.minutes
+  FETCH_REPLIES_INITIAL_WAIT_MINUTES = (ENV['FETCH_REPLIES_INITIAL_WAIT_MINUTES'] || 5).to_i.minutes
+
+  included do
+    scope :created_recently, -> { where(created_at: FETCH_REPLIES_INITIAL_WAIT_MINUTES.ago..) }
+    scope :not_created_recently, -> { where(created_at: ..FETCH_REPLIES_INITIAL_WAIT_MINUTES.ago) }
+    scope :fetched_recently, -> { where(fetched_replies_at: FETCH_REPLIES_COOLDOWN_MINUTES.ago..) }
+    scope :not_fetched_recently, -> { where(fetched_replies_at: [nil, ..FETCH_REPLIES_COOLDOWN_MINUTES.ago]) }
+
+    scope :should_not_fetch_replies, -> { local.or(created_recently.or(fetched_recently)) }
+    scope :should_fetch_replies, -> { remote.merge(not_created_recently).merge(not_fetched_recently) }
```

Comment:
I think the `merge` here is redundant, could just be written as `remote.not_created_recently` etc.

Reply:
Done in https://github.com/mastodon/mastodon/pull/32615/commits/f304bbd812e90780b8b6407b0a36cd07913d76dc